Running-Mastodon/Production-guide.md

```diff
@@ -1,20 +1,190 @@
-'X-Content-Type-Options' => 'nosniff',
-'X-XSS-Protection'       => '1; mode=block',
+
+Log in as the `mastodon` user:
+
+
+```sh
+su - mastodon
+```
+
+We will need to set up [`rbenv`](https://github.com/rbenv/rbenv) and [`ruby-build`](https://github.com/rbenv/ruby-build):
+
+```sh
+git clone https://github.com/rbenv/rbenv.git ~/.rbenv
+cd ~/.rbenv && src/configure && make -C src
+echo 'export PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.bashrc
+# Restart shell
+exec bash
+# Check if rbenv is correctly installed
+type rbenv
+# Install ruby-build as rbenv plugin
+git clone https://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build
+```
+
+Now that [`rbenv`](https://github.com/rbenv/rbenv) and [`ruby-build`](https://github.com/rbenv/ruby-build) are installed, we will install the
+[Ruby](https://github.com/rails/rails) version which [Mastodon](https://github.com/tootsuite/mastodon/) uses. That version will also need to be enabled.
+
+To enable [Ruby](https://github.com/rails/rails), run:
+
+```sh
+rbenv install 2.4.1
+rbenv global 2.4.1
+```
+
+**This will take some time. Go stretch for a bit and drink some water while the commands run.**
+
+### node.js And Ruby Dependencies
+
+Now that [Ruby](https://github.com/rails/rails) is enabled, we will clone the [Mastodon git repository](https://github.com/tootsuite/mastodon/) and install the [Ruby](https://github.com/rails/rails) and [node.js](https://nodejs.org/en/) dependancies.
+
+Run the following to clone and install:
+
+```sh
+# Return to mastodon user's home directory
+cd ~
+# Clone the mastodon git repository into ~/live
+git clone https://github.com/tootsuite/mastodon.git live
+# Change directory to ~live
+cd ~/live
+# Checkout to the latest stable branch
+git checkout $(git tag -l | sort -V | tail -n 1)
+# Install bundler
+gem install bundler
+# Use bundler to install the rest of the Ruby dependencies
+bundle install --deployment --without development test
+# Use yarn to install node.js dependencies
+yarn install --pure-lockfile
 ```
 
-## Nginx
+That is all we need to do for now with the `mastodon` user, you can now `exit` back to root.
 
-Regardless of whether you go with the Docker approach or not, here is an example Nginx server configuration.
+## PostgreSQL Database Creation
 
-At a minimum, you'll want to replace any occurrence of `example.com` with your actual hostname, and `/home/mastodon/live/public` with the location of your actual mastodon `public/` directory.
+[Mastodon](https://github.com/tootsuite/mastodon/) requires access to a [PostgreSQL](https://www.postgresql.org) instance.
+
+Create a user for a [PostgreSQL](https://www.postgresql.org) instance:
+
+```
+# Launch psql as the postgres user
+sudo -u postgres psql
+
+# In the following prompt
+CREATE USER mastodon CREATEDB;
+\q
+```
+
+**Note** that we do not set up a password of any kind, this is because we will be using ident authentication. This allows local users to access the database without a password.
```

Comment:
I wonder if we should continue advising people to set up Postgres without a password given that [PgBouncer doesn't seem to support it](https://github.com/tootsuite/documentation/blob/master/Running-Mastodon/PgBouncer-guide.md).